EchoPro/survey.py

```diff
@@ -363,6 +364,155 @@ def biometric_distributions( self ):
+        ### This applies the Jolly and Hampton (1990) stratified mean for transect survey designs
+        ### that provides a weighted mean and variance estimate for specified spatial regions (or other
+        ### similar strata definitions).
+        self.stratified_survey_statistics()
+        # NEW ATTRIBUTE: self.results
+        # OUTPUT: self.results['transect_results'] (dict)
+        ## self.results['transect_results']: {'mean': np.float64, 'cv': np.float64}
+        ### These may be pd.DataFrame instead of np.float64 to allow for grouped calculations (e.g. 
+        ### by sex, age, etc)
+        ## this would provide the coefficient of variation, but the actual variance output 
+        ## is largely arbitrary since it is simply normalized by the mean
+        ### The name of this may be different -- it should be differentiated from the kriged 
+        ### results -- perhaps something like "nominal_results" or something
+        
+    @staticmethod
+    def kriging_analysis():
+        
+        # Organize semivariogram and kriging parameters 
+        ... = self.statistics['kriging']['vario_krig_para_df']
+        ### TODO: Perhaps separate 'vario.__' and 'krig.__' ?
+        
+        # Prepare kriging mesh parameterization
+        ### This is necessary for ensuring that all required parameters
+        ### are mapped to each node within the kriging mesh
+        ### This would replace 'bin_dataset()' in the previous implementation
+        self.initialize_kriging_mesh()    
+        
+        # Fit semivariogram 
+        self.fit_semiovariogram_model()
+        # OUTPUT: self.statistics['semivariogram'] (dict)
+        ### Keys represent each specific model parameter such as the 
+        ### range, sill, nugget, etc.
+        
+        # Apply semiovariogram to interpolate data over the defined kriging mesh
+        self.kriging_interpolation()
+        # OUTPUT: self.statistics['kriging']['modeled_biomass'] (df)
+        ## Or perhaps this would also be appropriate under the 'biology' attribute
+                
+        # Calculate similar stratified survey analysis but this time using
+        # kriged values
+        self.stratified_kriging_statistics()
+        # OUTPUT: self.results['kriging_results'] (dict)
+        ## self.results['kriging_results']: {'mean': np.float64, 'cv': np.float64}
+        ### These may be pd.DataFrame instead of np.float64 to allow for grouped calculations (e.g. 
+        ### by sex, age, etc)
+        
+    @staticmethod
+    def strata_mean_sigma_bs():
+        
+        # Calculate sigma_bs from TS-length regression
+        self.TS_L_regression( self.biology['length_df'] , self.biology['specimen_df'])
+        
+        # Calculate mean value per stratum here 
+        
+    def impute_missing_strata():
+        
+    def fit_length_weight_relationship( length_df , specimen_df ):
+        
+    
+        
+    @staticmethod
+    def TS_L_regression( length_df , specimen_df ):
```

Comment:
I know this is just a sketch, so just a small note that if this is not a static method then `length_df` and `specimen_df` are accessible through `self`.